manifests/init.pp

```diff
@@ -30,32 +40,53 @@
 #   *If* the $servers array above starts with '127.0.0.1' or '::1', then
 #   the system will set itself up as a caching nameserver unless this is set
 #   to false.
-#
+# @param sortlist
+#   Optional Array of address/netmask pairs that allow addresses returned by
+#   gethostbyname to be sorted.
+# @param extra_options
+#   Optional Array to put any options that may not be covered by the variables
+#   below. These will be appended to the options string.
 class resolv (
   Simplib::Netlist           $servers        = simplib::lookup('simp_options::dns::servers', { 'default_value' => ['127.0.0.1'] }),
   Simplib::Netlist           $search         = simplib::lookup('simp_options::dns::search', { 'default_value'  => [$facts['domain']] }),
-  Optional[Simplib::Netlist] $sortlist       = undef,
-  Optional[Array]            $extra_options  = undef,
-  String                     $resolv_domain  = $facts['domain'],
+  Resolv::Domain             $resolv_domain  = $facts['domain'],
```

Comment:
Should we be handling domain and search differently? From the man page:

> The domain and search keywords are mutually exclusive.  If  more  than one instance of these keywords is present, the last instance wins.

Reply:
For extra credit, we could add a helpful warning or comment if content is supplied for both parameters, but their inclusion is valid syntax, and is handled by a known and documented behavior.